libusb/os/windows_common.h

```diff
@@ -232,12 +232,29 @@ typedef struct USB_DK_TRANSFER_REQUEST {
 	USB_DK_TRANSFER_RESULT Result;
 } USB_DK_TRANSFER_REQUEST, *PUSB_DK_TRANSFER_REQUEST;
 
+/* track devices that are removed, kept unchanged, or added when updating device list
+ * using system enumeration, in response to low-level hotplug related events */
+ enum enumeration_status {
+	/* device is no longer present in system enumeration.
+	 * A hotplug "device left" event shall be triggered and the device removed from the list */
+	NO_LONGER_ENUMERATED,
+
+	/* device is still in system enumeration as it was in previous enumeration.
+	 * There is no need to take any action */
+	STILL_ENUMERATED,
+
+ 	/* device is newly added to the list, meaning it just appeared in system enumeration.
+ 	 * A hotplug "device arrived" event shall be triggered */
+	NEWLY_ENUMERATED
+};
+
 struct usbdk_device_priv {
 	USB_DK_DEVICE_ID ID;
 	PUSB_CONFIGURATION_DESCRIPTOR *config_descriptors;
 	HANDLE redirector_handle;
 	HANDLE system_handle;
 	uint8_t active_configuration;
+	enum enumeration_status discovery_status; /* updated while getting current device list */
```

Comment:
Maybe use the same comment suggest for `winusb_device_priv`?

Reply:
ok